sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1207,6 +1207,12 @@ class Analyzer(
                 val alias = Alias(ae, ae.toString)()
                 aggregateExpressions += alias
                 alias.toAttribute
+              case ne: NamedExpression => ne
```

Comment:
Removing it will cause error in `replaceGroupingFunc`. As we create a new attribute based on the grouping column with additional alias, we can't find it anymore in aggregate's group by expressions.